### PR TITLE
DBZ-3605 Add support for several Oracle data types

### PR DESCRIPTION
**Pull request by Naros**
https://issues.redhat.com/browse/DBZ-3605

Data Types:
- [x] XMLTYPE
- [x] RAW
- [ ] ~~LONG - Not likely supported due to complications with LogMiner.~~
- [ ] ~~LONG RAW - Not likely supported due to the same complications with LogMiner as LONG types.~~
- [x] Documentation
- [x] Fix dependencies on XDB and SAX environment variables

### CODE REVIEW

**Comment by Naros** [comment]
So `LONG` (replaced by CLOB) and `LONG RAW` (replaced by BLOB) are data types provided by LogMiner in two different ways depending on the data length in those columns.  The use case where this becomes problematic is when these data types are updated with a new value.

If the length of the character data (`LONG`) or byte data (`LONG RAW`) is short enough, Oracle inlines the column values in the re-constructed SQL so the SQL statements appear as follows:

```
UPDATE table SET long_col = 'the long text' WHERE long_col = 'the old long text value';
UPDATE table SET long_raw_col = HEXTORAW('new hex value') WHERE long_raw_col = HEXTORAW('old hex value');
```

This works well, but unfortunately, once the character or binary data length reaches a specific threshold, Oracle will re-construct the SQL differently, using what is called `LONG DATA` operations (or opcode 24) event types.  The event stream becomes the following:

```
UPDATE table set long_col = 'LONG DATA' WHERE long_col = 'LONG DATA';
[a series of LONG DATA events]
```
or
```
UPDATE table set long_raw_col = 'LONG DATA' WHERE long_raw_col = 'LONG DATA';
[a series of LONG DATA events]
```

These `LONG DATA` operation events are a continuous stream of character or binary data representing both the old and new values.  For `LONG RAW`, we can only determine the breakpoint between the old/new values because the stream is the concatenation of `HEXTORAW('...')HEXTORAW('...')` where the former function call is old, and the latter is new; however when working with `LONG` data types, this distinction doesn't exist.

The only reason that `RAW` works is that it is explicitly restricted to a maximum length of 2000, which allows Oracle to inline the value, and therefore its always directly available in the re-constructed SQL without any secondary event types that require merge semantics.

We can safely support `LONG` and `LONG RAW` via Xstream because GoldenGate can differentiate the break point in the `LONG DATA` redo entries, but unfortunately LogMiner does not.  This means that the OpenLogReplicator implementation should be more than capable of doing precisely what GoldenGate does for Xstream, so if we wanted to pursue this support, it would be non-LogMiner only for these two.

**Comment by Naros** [comment]
@roldanbob could you take a peak at the recent commit for the documentation changes, and suggest or add whatever changes are needed for the product side around the new XDB library dependency?

File: documentation/modules/ROOT/pages/connectors/oracle.adoc
Comment (roldanbob) [inline documentation/modules/ROOT/pages/connectors/oracle.adoc:1418]:
```suggestion
|Depending on the setting of the xref:oracle-property-binary-handling-mode[`binary.handling.mode`] property in the connector configuration, the connector maps LOB values of this type to one of the following the semantic types:

* Raw bytes (the default)
* A base64-encoded string
* A base64-url-safe-encoded string
* A hex-encoded string.
```

File: documentation/modules/ROOT/pages/connectors/oracle.adoc
Comment (roldanbob) [inline documentation/modules/ROOT/pages/connectors/oracle.adoc:1745]:
```suggestion

[id="oracle-xml-types"]
=== XML types
```

File: documentation/modules/ROOT/pages/connectors/oracle.adoc
Comment (roldanbob) [inline documentation/modules/ROOT/pages/connectors/oracle.adoc:1749]:
```suggestion
Support for `XMLTYPE` is currently in an incubating state.
Depending on the feedback that we receive, the exact semantics, configuration options, and so forth might change in future revisions.
```

File: documentation/modules/ROOT/pages/connectors/oracle.adoc
Comment (roldanbob) [inline documentation/modules/ROOT/pages/connectors/oracle.adoc:1756]:
```suggestion
Use of the `XMLTYPE` with the {prodname} Oracle connector is a Technology Preview feature only.
```

File: documentation/modules/ROOT/pages/connectors/oracle.adoc
Comment (roldanbob) [inline documentation/modules/ROOT/pages/connectors/oracle.adoc:2154]: In your review request, you refer to changes that might be "needed for the product side", which I take to mean for use downstream. But this content is only in the Deployment topic that's conditionalized for community use. 

Do we need to include a reference to obtaining the XDB library in `/partials/modules/all-connectors/ref-deploy-oracle-kafka-connect-yaml.adoc` ([L10](https://github.com/debezium/debezium/blame/main/documentation/modules/ROOT/partials/modules/all-connectors/ref-deploy-oracle-kafka-connect-yaml.adoc#L10), and [L42](https://github.com/debezium/debezium/blame/main/documentation/modules/ROOT/partials/modules/all-connectors/ref-deploy-oracle-kafka-connect-yaml.adoc#L42))?

And probably a short topic in the Deployment overview in `oracle.adoc`, analogous to (and adjacent to)  _[Obtaining the Oracle JDBC driver_,](https://github.com/debezium/debezium/blame/main/documentation/modules/ROOT/pages/connectors/oracle.adoc#L2151) that explains why you might want to obtain the XDB library.

File: documentation/modules/ROOT/pages/connectors/oracle.adoc
Comment (roldanbob) [inline documentation/modules/ROOT/pages/connectors/oracle.adoc:1767]:
```suggestion
|Oracle data type
```
We use sentence-style capitalization in table headings downstream. I know that this isn't consistently applied upstream, so we can either let it go here for now, or make the change and give me the task to do a thorough scrubbing of the doc set to apply the formatting.

File: documentation/modules/ROOT/pages/connectors/oracle.adoc
Comment (roldanbob) [reply documentation/modules/ROOT/pages/connectors/oracle.adoc:1418]: We could apply a similar change to the description on L1398.

**Review by roldanbob — CHANGES_REQUESTED** [review]

File: documentation/modules/ROOT/pages/connectors/oracle.adoc
Comment (Naros) [reply documentation/modules/ROOT/pages/connectors/oracle.adoc:2154]: That's correct @roldanbob.  Additionally the user will need to obtain the `xmlparserv2` dependency at 
```
<dependency>
  <groupId>com.oracle.database.xml</groupId>
  <artifactId>xmlparserv2</artifactId>
  <version>21.6.0.0</version>
</dependency>
```
I also added a blurb about the `-Djavax.xml.parsers.SAXParserFactory` argument required for XML support with Oracle, so we may need to rework how to include this elsewhere when enabling XML support.

Comment (roldanbob) [reply documentation/modules/ROOT/pages/connectors/oracle.adoc:2154]: Agreed. A procedure module, e.g., _Enabling support for Oracle XML data types_.

As far as adding content to the downstream deployment instructions, how shall we handle that.  Want me to open a separate PR against DBZ-3605 to make those changes? Or I could wait until you merge this and then add the downstream version of the content?

**Comment by jpechane** [comment]
@Naros Applied, thanks!

**Comment by Naros** [comment]
@roldanbob I'll follow-up the documentation changes in a follow PR later this week.

**Pull request merged**
